tutorials/preprocessing/35_artifact_correction_regression.py

```diff
@@ -23,146 +24,195 @@
-
-Note that there are other corrections that are useful for this dataset that we
-will not apply here (see :ref:`tut-brainstorm-auditory` for more information).
+data. The :ref:`MNE-Sample <sample-dataset>` dataset has some clear, large
+blink artifacts, especially during the presentation of visual stimuli.
 """
 
 # %%
-
-import os.path as op
 import numpy as np
 import mne
+from mne.preprocessing import EOGRegression
+
+data_path = mne.datasets.sample.data_path()
+raw_fname = data_path / 'MEG' / 'sample' / 'sample_audvis_raw.fif'
+raw = mne.io.read_raw_fif(raw_fname)
+raw.pick(['eeg', 'eog', 'stim'])
+raw.load_data()
+
+# The regression technique works regardless of chosen reference. However, it is
+# important to choose a reference before proceeding with the analysis.
+raw.set_eeg_reference('average')
+raw.apply_proj()
+
+# Removing slow drifts makes for more stable regression coefficients. Make sure
+# to apply the same filter to both EEG and EOG channels!
+raw.filter(0.3, 40)
+
+# make epochs
+events = mne.find_events(raw)
+event_id = {'visual/left': 3, 'visual/right': 4}
+epochs = mne.Epochs(raw, events, event_id=event_id, preload=True)
+
+# %%
+# Visualize the original data
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Let's first look at the `~mne.Evoked` data (average across epochs) without
+# any corrections applied.
 
-data_path = mne.datasets.brainstorm.bst_auditory.data_path()
-raw_fname = op.join(data_path, 'MEG', 'bst_auditory', 'S01_AEF_20131218_01.ds')
-raw = mne.io.read_raw_ctf(raw_fname).crop(0, 60)
-raw.set_channel_types({'HEOG': 'eog', 'VEOG': 'eog', 'ECG': 'ecg'})
-raw.pick_types(meg=True, stim=True, misc=True,
-               eog=True, ecg=True, ref_meg=False).load_data()
-raw.info['bads'] = ['MLO52-4408', 'MRT51-4408', 'MLO42-4408', 'MLO43-4408']
-raw.filter(None, 40)
-decim = 12  # 2400 -> 200 Hz sample rate for epochs
+# we'll try to keep a consistent ylim across figures
+plot_kwargs = dict(picks='all', ylim=dict(eeg=(-10, 10), eog=(-5, 15)))
+
+# plot the evoked for the EEG and the EOG sensors
+fig = epochs.average('all').plot(**plot_kwargs)
+fig.set_size_inches(6, 6)
 
 # %%
-# For this dataset and example we'll use just event type 1 (only the
-# "standard" trials) for simplicity. Event timing is adjusted by comparing the
-# trigger times on detected sound onsets on channel UADC001-4408.
-
-events = mne.find_events(raw, stim_channel='UPPT001')
-sound_data = raw[raw.ch_names.index('UADC001-4408')][0][0]
-onsets = np.where(np.abs(sound_data) > 2. * np.std(sound_data))[0]
-min_diff = int(0.5 * raw.info['sfreq'])
-diffs = np.concatenate([[min_diff + 1], np.diff(onsets)])
-onsets = onsets[diffs > min_diff]
-assert len(onsets) == len(events)
-events[:, 0] = onsets
-epochs = mne.Epochs(raw, events, event_id=1, decim=decim, preload=True)
+# We can see there is some EOG activity that is likely bleeding into the EEG
+# evoked response. At around 250ms this becomes especially noticeable. Let's
```

Comment:
There should be a non-breaking space between the number and its unit. I don't know if this is supported here, I would use either `&nbsp;` or `&thinsp;` (preferably the latter). This should be consistent throughout our documents.

Reply:
I don't think we do this anywhere in the documentation. Sounds like a good idea though. We should make a `unit` macro or something.

Reply:
Good idea, can you open an issue so that we don't forget?

Reply:
#11208